pkg/client/events/replay_client.go

```diff
@@ -142,22 +147,27 @@ func (rClient *replayClient[T, R]) EventsSequence(ctx context.Context) R {
 // goRemapEventsSequence publishes events observed by the most recent cached
 // events type replay observable to the given publishCh
 func (rClient *replayClient[T, R]) goRemapEventsSequence(ctx context.Context, publishCh chan<- T) {
-	// Subscribe to the replayObsCache to listen for changes to the "active"
-	// replay observable.
-	cachedEventTypeObserver := rClient.replayObsCache.Subscribe(ctx)
-	for eventObs := range cachedEventTypeObserver.Ch() {
-		select {
-		case <-ctx.Done():
-			return
-		default:
-		}
-		// Publish events from the new "active" replay observable to the given
-		// publish channel.
-		eventObserver := eventObs.Subscribe(ctx)
-		for event := range eventObserver.Ch() {
-			publishCh <- event
-		}
-	}
+	var prevEventTypeObs observable.ReplayObservable[T]
+	channel.ForEach[R](
+		ctx,
+		rClient.replayObsCache,
+		func(ctx context.Context, eventTypeObs R) {
+			if prevEventTypeObs != nil {
+				// Just in case the assumption that all transport errors are
```

Comment:
`Just in case the assumption that all transport errors are persistent does not hold`

I'm not sure what this means. #PUC